pandas/tools/pivot.py

```diff
@@ -439,8 +439,15 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     crosstab : DataFrame
     """
 
-    index = com._maybe_make_list(index)
-    columns = com._maybe_make_list(columns)
+    def _make_list(arr):
+        # see gh-12298
+        if com.is_categorical(arr):
+            arr = Series(list(arr), name=arr.name)
```

Comment:
There should be a more efficient way to do that (than convert to list in between), by using the dtype of the categories.